pkg/services/grafana-apiserver/storage/entity/utils.go

```diff
@@ -0,0 +1,263 @@
+package entity
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	"github.com/grafana/grafana/pkg/infra/grn"
+	"github.com/grafana/grafana/pkg/kinds"
+	entityStore "github.com/grafana/grafana/pkg/services/store/entity"
+)
+
+type Key struct {
+	Group       string
+	Resource    string
+	Namespace   string
+	Name        string
+	Subresource string
+}
+
+func ParseKey(key string) (*Key, error) {
+	// /<group>/<resource>/<namespace>/<name>(/<subresource>)
+	parts := strings.SplitN(key, "/", 6)
+	if len(parts) != 5 && len(parts) != 6 {
+		return nil, fmt.Errorf("invalid key (expecting 4 or 5 parts) " + key)
+	}
+
+	if parts[0] != "" {
+		return nil, fmt.Errorf("invalid key (expecting leading slash) " + key)
+	}
+
+	k := &Key{
+		Group:     parts[1],
+		Resource:  parts[2],
+		Namespace: parts[3],
+		Name:      parts[4],
+	}
+
+	if len(parts) == 6 {
+		k.Subresource = parts[5]
+	}
+
+	return k, nil
+}
+
+func (k *Key) String() string {
+	if len(k.Subresource) > 0 {
+		return fmt.Sprintf("/%s/%s/%s/%s/%s", k.Group, k.Resource, k.Namespace, k.Name, k.Subresource)
+	}
+	return fmt.Sprintf("/%s/%s/%s/%s", k.Group, k.Resource, k.Namespace, k.Name)
+}
+
+func (k *Key) IsEqual(other *Key) bool {
+	return k.Group == other.Group &&
+		k.Resource == other.Resource &&
+		k.Namespace == other.Namespace &&
+		k.Name == other.Name &&
+		k.Subresource == other.Subresource
+}
+
+func (k *Key) TenantID() (int64, error) {
+	if k.Namespace == "default" {
+		return 1, nil
+	}
+	tid := strings.Split(k.Namespace, "-")
+	if len(tid) != 2 || !(tid[0] == "org" || tid[0] == "tenant") {
```

Comment:
`stack-` is used in [existing namespace parsing](https://github.com/grafana/grafana/blob/38bc41651a416bd547adbb284bc1559a5aad1006/pkg/services/grafana-apiserver/endpoints/request/namespace.go#L68). should one of them be changed?